### ISSUE TITLE
Import declaration conflicts with local declaration

### ISSUE THREAD

**Issue by jeffpyke** [stephenh/ts-proto#36]
file1.proto
```proto
syntax = "proto2";

package common.a;
message Foo {}
```
file2.proto
```proto
syntax = "proto2";
import "file1.proto";

package common.b;
message Foo {}

message Bar {
  optional common.a.Foo foo = 1;
}
```

Generated Typescript (file2.ts) (error):
```ts
import { Foo } from 'file1';
export interface Foo {
}
export interface Bar {
  foo: Foo | undefined;
}
```
I think this error is caused by loss of the package information.

**Comment by stephenh** [comment]
Thanks for the report; kind of surprised b/c ts-proto generally tries to understand packages, although the mapping between a proto package and JS/TS module and file paths are all kind of murky. I'll try to take a look at this soon. I might get distracted reorganizing some of the test projects to make this easier to repro...

**Comment by stephenh** [comment]
Oh wait, I see. I missed that local `message Foo` when I first scanned this issue this morning.

Yeah, that will be a little bit more involved.

In theory, the ts-poet library that ts-proto uses is supposed to solve exactly this sort of "fancy auto-import" type thing, but it doesn't realize there could be local vs. remote conflicts.

So, ideally getting ts-poet fixed would transitively fix this issue, although then there is a small wrinkle that ts-poet has a 3.x API out that is a fairly big breaking change from the 2.x API that ts-proto is currently using.

Basically, yeah, I see the issue, and will (likely slowly but hopefully steadily) work towards a fix.

**Comment by stephenh** [comment]
As an update here, I implemented import collision avoidance in the latest/3.x API for ts-poet, see this test case:

https://github.com/stephenh/ts-poet/blob/master/src/index-tests.tsx#L163

Which is cool, but now means either: a) backporting that feature to the ts-poet 2.x branch that ts-proto uses (...probably easier?), or b) migrating ts-proto onto ts-poet 3.x (best for the long term and should not be hard but definitely tedious work).

If this is a big pain point/blocker for you, you could look into either of those, otherwise I'll get around to b) at some point.

**Comment by jeffpyke** [comment]
Thanks, that's neat! It's not a huge pain point right now, because we can just delete the (unused) conflicting import manually. Looking forward to the fix! I think it will be useful once we generate a larger number of proto ts interfaces (right now 10-20), but eventually have 1000s of proto files in a mono repo we would like to generate ts interfaces for. Thanks, this project is great!

**Comment by stephenh** [comment]
Hey @jeffpyke , so it took a few months, but I got around to the ts-poet upgrade, so this should be fixed now in v1.47.0. Let me know if you still see issues. Thanks!

**Comment by stezu** [comment]
@stephenh I'm having a similar issue with the latest version of `ts-proto`. I created [a sample repo](https://github.com/stezu/ts-proto-bug) to help explain. I have a proto file which imports messages from two other proto files which happen to have the same name. `ts-proto` (and by extension `ts-poet`) don't avoid the naming conflict so I end up with the following in my output TS file:
```ts
import { User } from "../protos/auth/types";
import { User } from "../protos/user/types";
```

I'll spend some time digging into the issue for a good place to resolve it, but if you have any pointers, they would be greatly appreciated!

**Comment by stephenh** [comment]
@stezu okay, try ts-proto v1.56.0, and see if that fixes it. Thanks!
